src/boot/efi/boot.c

```diff
@@ -101,6 +101,53 @@ enum {
         TIMEOUT_TYPE_MAX    = UINT32_MAX,
 };
 
+#define _returns_nonnull_ __attribute__((__returns_nonnull__))
+#define _allocate_pool_ __attribute__((__malloc__, __malloc__(FreePool, 1), __warn_unused_result__))
+#define assert_alloc_return(p) ({ \
+        if (_likely_(p))          \
+                return p;         \
+        log_oom();                \
+        __builtin_trap();         \
```

Comment:
uh, not sure what that trap would do in uefi mode.

we should just freeze for good here

Reply:
> GCC implements this function by using a target-dependent mechanism (such as intentionally executing an illegal instruction) or by calling abort.
Since it's -ffreestanding, it's most likely an invalid op. But this was prototype code anyway.